index.js

```diff
@@ -4,19 +4,24 @@ var node_modules = "node_modules"
 
 function Found() {}
 
-function search(dir) {
+function search(dir, scope) {
   dir = dir || "."
   var context = path.resolve(dir)
   if (!fs.existsSync(context)) return []
   var contents = fs.readdirSync(context)
-  return contents.map(function(name) {
+  return contents.reduce(function (accumulated, name) {
+    if (name === ".bin") return accumulated; // .bin directory is never a package
```

Comment:
`.bin` would get filtered out anyway by `is` right? Just wondering is there another reason to include?

Reply:
@ryanve I don't feel very strong about this one so I'm dropping it for now. Believe that `is` step is running both `fs.existsSync` and `fs.lstatSync` which is touching the file system twice unnecessarily and in the weird instance where `.bin` is a link, I think the results would be pretty wacky since it may interpret it as a package (thinking of alternate npm clients that may do things in unexpected ways).

On side note, I think `is` would be better off as:
```js
function is(p) {
  if (p instanceof Found) return p.hasOwnProperty("link")
  try {
    return fs.lstatSync(p).isSymbolicLink()
  } catch (err) {
    return false
  }
}
```

Reasons are that this makes the operation atomic, faster (less reads), and the majority cases where it wouldn't exist are exceptional. Node's primary API deprecated `fs.exists` long ago for these reasons - https://nodejs.org/api/fs.html#fs_fs_exists_path_callback

Also, seems like `is` might be better off as two separate functions `isLink` and `hasLink`, given that its doing completely different things depending on the argument - having it all in one made debugging more difficult for me as a noob. `$.02` 😄

Reply:
Good info. I can do a follow-up to address that. I did the `is` function like that to be terse but `has` could be separate. This was the reason for doing `var found = new Found` instead of using a plain object `var found = {}`